vmdb/spec/requests/api/roles_spec.rb

```diff
@@ -0,0 +1,247 @@
+#
+# Rest API Request Tests - Roles specs
+#
+# - Creating a role                       /api/roles                            POST
+# - Creating a role via action            /api/roles                            action "create"
+# - Creating multiple roles               /api/roles                            action "create"
+# - Edit a role                           /api/roles/:id                        action "edit"
+# - Edit multiple roles                   /api/roles                            action "edit"
+# - Delete a role                         /api/roles/:id                        DELETE
+# - Delete a role by action               /api/roles/:id                        action "delete"
+# - Delete multiple roles                 /api/roles                            action "delete"
+#
+require 'spec_helper'
+
+describe ApiController do
+  include Rack::Test::Methods
+
+  let(:feature_identifiers) { ["vm_explorer" ,"ems_infra_tag" ,"my_settings_time_profiles",
+                          "miq_request_view", "miq_report_run", "storage_manager_show_list"] }
+  let(:expected_attributes) { %w(id name read_only settings) }
+  let(:sample_role1) do
+    {
+      "name"      => "sample_role_1",
+      "settings"  => { "restrictions"  => { "vms" => "user" } },
+      "miq_product_features"  => [
+          { :identifier => "vm_explorer" },
+          { :identifier => "ems_infra_tag" },
+          { :identifier => "my_settings_time_profiles" }
+      ]
+    }
+  end
+  let(:sample_role2) do
+    {
+      "name"      => "sample_role_2",
+      "settings"  => { "restrictions"  => { "vms" => "user_or_group" } },
+      "miq_product_features"  => [
+          { :identifier => "miq_request_view" },
+          { :identifier => "miq_report_run" },
+          { :identifier => "storage_manager_show_list" }
+      ]
+    }
+  end
+  let(:sample_role3) do
+    {
+      "name"      => "sample_role_3"
+    }
+  end
+
+  before(:each) do
+    init_api_spec_env
+    features = feature_identifiers.collect do |identifier|
+      MiqProductFeature.find_or_create_by_identifier(identifier)
+    end
+  end
+
+  def app
+    Vmdb::Application
+  end
+
```

Comment:
please add role queries of features subcollection here.   primary roles collection is tested in collections_spec.rb.